### DISCUSSION TITLE
Remap using txt: vs a normal image gets very different results.

### DISCUSSION THREAD

**Discussion by Dakedres** [ImageMagick/ImageMagick#7180] [Help]
### ImageMagick version

7.1.1-29

### Operating system, version and so on

Void Linux, kernel 6.6.21_1

### Description

I'm noticing very different results when attempting to dither an image using a pixel enumeration file (`txt:`) vs. a binary image.
[Source image](https://github.com/ImageMagick/ImageMagick/assets/36781227/ff1cc0bf-1a38-4b34-8c06-f3cb3146d832)

Using pixel enumeration:
```
convert source.jpg -dither FloydSteinberg -remap txt:palette.txt out.png
```
![out](https://github.com/ImageMagick/ImageMagick/assets/36781227/2854d197-c266-43b3-8a22-42fdf1b6f2f7)

Using a .png palette file (generated from the same pixel enumeration using `convert txt:palette.txt palette.png`)
```
convert source.jpg -dither FloydSteinberg -remap txt:palette.png out.png
```
![out](https://github.com/ImageMagick/ImageMagick/assets/36781227/765685fd-d969-49a4-b1ad-8640ede7360c)

A clearly better result.

I imagine this has something to do with the fact that the pixel enumeration doesn't really have a color index to pull from, so ImageMagick has to make do in some way, but I'd really like if there was some way to do this without creating an extra file. In my original use case I was generating the pixel enumeration with another program and piping it directly into imagemagick.
I'm also just generally curious what's going on here and would like to learn more.

Thank you!

**Comment by Dakedres** [comment]
I also notice that different file formats for the palette give very different effects, too. Even using a palette.jpg differs a lot from PNG

**Comment by snibgo** [comment]
Please show your palette.txt.

**Comment by Dakedres** [comment]
```
# ImageMagick pixel enumeration: 16,1,255,rgb
0,0: (20,31,46)  #141F2E  base00
1,0: (30,92,30)  #1e5c1e  base01
2,0: (39,62,92)  #273e5c  base02
3,0: (160,255,160)  #a0ffa0  base03
4,0: (30,92,30)  #1e5c1e  base04
5,0: (255,204,51)  #FFCC33  base05
6,0: (145,204,255)  #91CCFF  base06
7,0: (55,87,128)  #375780  base07
8,0: (255,250,177)  #fffab1  base08
9,0: (255,128,128)  #FF8080  base09
10,0: (145,204,255)  #91CCFF  base0A
11,0: (128,255,128)  #80ff80  base0B
12,0: (128,255,128)  #80ff80  base0C
13,0: (162,207,245)  #a2cff5  base0D
14,0: (0,153,255)  #0099FF  base0E
15,0: (231,231,231)  #e7e7e7  base0F
```

**Comment by snibgo** [comment]
> # ImageMagick pixel enumeration: 16,1,255,rgb

Your palette text file is encoded as rgb. Your source file is encoded as sRGB. That is the problem: remapping an image with another in a different colorspace.

When combining images with any operation, it is important that the user ensures they are in the same colorspace.
